src/entity/Company.java

```diff
@@ -0,0 +1,30 @@
+package entity;
+
+import utils.CarPoolGenerator;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class Company {
+    private List<Car> cars;
+    private static Company company = null;
+    private static AtomicBoolean instanceCreated = new AtomicBoolean(false);
+
+    private Company() {
+        cars = CarPoolGenerator.getCars();
+    }
+
+    public static Company getCompany() {
+
+        if (company == null && !instanceCreated.get()) {
```

Comment:
You don't need to check _null_ by due to checking **instanceCreated** variable or check null inside first if statement